### PR TITLE
refactor: make tts config more understandable and easier

### PR DESCRIPTION
**Pull request by Maxsiom**
### Description

The TTS config now only has two instead of three booleans to manage caching.
It now always saves the cached audio files in StreamingAssets (there was no alternative before)
and the users can decide whether they want to also cache during runtime when the application
was already built or not.
This is useful when the TTS provider is not always available (e.g. no Internet connection).
Android cannot write into StreamingAssets, so this should be deactivated when building for Android.

**Fixes**: TTS config was not really understandable.

### Type of change

- [x] Refactor with obsoletes

### How Has This Been Tested?

Tested on my machine. Also loaded older projects to check.

### Checklist

- [x] My code follows the [Coding Conventions](#coding-conventions)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Maxsiom** [comment]
Yes, and yes. 
Because `StillWriteCacheWhenProjectIsBuilt` writes into Streaming Assets and Android cannot write into Streaming Assets. :)
But you always want to cache when being in editor. So therefore is the `UseCache` boolean. On Windows you can also cache in built applications. So you could enable `StillWriteCacheWhenProjectIsBuilt`.

**Comment by Maxsiom** [comment]
Before `SaveAudioFilesToStreamingAssets` and `UseStreamingAssetFolder` were pretty much the same and Simon and I did not understand the difference.

**Comment by Gusinuhe** [comment]
> Before `SaveAudioFilesToStreamingAssets` and `UseStreamingAssetFolder` were pretty much the same and Simon and I did not understand the difference.

Yeah, me neither 😕
I would suggest easing the complexity in the configuration currently, it seems that we need a configuration for windows and another for Android in order to do the same.

Take a look into the `FileManager`, you could find some utils to help you deal with the `StreammingAssets` and `PersistentDataPath` folders for when playing in the editor or in the built app.

I think the settings could be reduced to a simply `UseCache` for all platforms.

**Comment by Maxsiom** [comment]
Luckily this is also not needed for the bug fix. So we have time to figure this out. 👍

**Comment by SimonTheSourcerer** [comment]
Using UseCache is already an improvement, we still should allow to disable writing the cache in a build version. We probably also should have this option on false, which allows to work on both, android and windows

File: Runtime/TextToSpeechConfiguration.cs [hunk 1]
```diff
@@ -24,24 +25,31 @@ public class TextToSpeechConfiguration : ScriptableObject
...
-        /// Usage of the standard HTML cache.
-        /// </summary>
-        public bool UseCache = true;
-
         /// <summary>
         /// Enables the usage of the streaming asset folder as second cache directory to allow deliveries which work offline too.
         /// This means the StreamingAssets/{StreamingAssetCacheDirectoryName} will be searched first and only if there
         /// is no fitting audio file found the text to speech provider will be used to download the audio file from web.
         /// </summary>
+        public bool UseCache = true;
```
Comment (aleksei-korolev) [inline Runtime/TextToSpeechConfiguration.cs:33]: Would be a break if the users had UseStreamingAssetFolder set to false.

File: Runtime/TextToSpeechConfiguration.cs [hunk 2]
```diff
@@ -24,24 +25,31 @@ public class TextToSpeechConfiguration : ScriptableObject
...
+        /// With this option enabled the application tries to save/cache all downloaded audio files to
+        /// StreamingAssets/{StreamingAssetCacheDirectoryName} during runtime when it is already built.
+        /// Should not be used when building for Android.
         /// </summary>
+        public bool StillWriteCacheWhenProjectIsBuilt = false;
+
+        /// <inheritdoc cref="StillWriteCacheWhenProjectIsBuilt"/>
+        [Obsolete("Use StillWriteCacheWhenProjectIsBuilt instead.")]
+        [HideInInspector]
         public bool SaveAudioFilesToStreamingAssets = false;
```
Comment (aleksei-korolev) [inline Runtime/TextToSpeechConfiguration.cs:50]: Same, but when user had set it to true.

File: Runtime/Provider/FileTextToSpeechProvider.cs [hunk 3]
```diff
@@ -52,7 +52,7 @@ public async Task<AudioClip> ConvertTextToSpeech(string text)
             {
                 audioClip = await FallbackProvider.ConvertTextToSpeech(text);
 
-                if (Configuration.SaveAudioFilesToStreamingAssets)
+                if (Configuration.UseCache && (Configuration.StillWriteCacheWhenProjectIsBuilt || Application.isEditor))
```
Comment (aleksei-korolev) [inline Runtime/Provider/FileTextToSpeechProvider.cs:55]: `StillWriteCacheWhenProjectIsBuilt` sounds like "Write cache when Unity builds the project even if Use Cache is toggled off". "UseCacheInBuiltApp" sounds better. Also, we can check for the current OS/build target on our own, so we could drop this setting, too.

**Pull request closed without merging**